pkg/operator/starter.go

```diff
@@ -117,6 +116,28 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 	kubeInformersNamespaced.Start(stopChan)
 	kubeInformersForNamespaces.Start(stopChan)
 
+	// Poll every minute to ensure removal of 4.3 deployments that were replaced in 4.4
+	// with a unified deployment.
+	//
+	// This code can be removed in 4.5 since downgrade to 4.3 will no longer be possible.
+	deployClient := kubeClient.AppsV1().Deployments(operatorclient.TargetNamespace)
+	go wait.Until(func() {
+		// List rather than blindly deleting so that there are max 1 request/minute
+		// instead of 3 (one for each 4.3 deployment).
+		deploys, err := deployClient.List(metav1.ListOptions{})
+		if err != nil {
+			klog.Warningf("Failed to list deployments when searching for 4.3 deployments to remove: %v", err)
+		}
```

Comment:
return.  `deploys` could be nil leading to a crash in the next line.

Reply:
Yikes, good catch. Apparently I'm nothing without tests!